internal/agent/device/lifecycle/manager.go

```diff
@@ -92,12 +102,85 @@ func (m *LifecycleManager) Initialize(ctx context.Context, status *v1alpha1.Devi
 }
 
 func (m *LifecycleManager) Sync(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	// this controller currently does not implement a sync operation
 	return nil
 }
 
 func (m *LifecycleManager) AfterUpdate(ctx context.Context, current, desired *v1alpha1.RenderedDeviceSpec) error {
-	// TODO: implement
+	if current.Decommission == nil && desired.Decommission != nil {
+		m.log.Warn("Detected decommissioning request from flightctl service")
+		if err := m.updateWithStartedCondition(ctx); err != nil {
+			return fmt.Errorf("failed to update decommission started status: %w", err)
+		}
+		m.log.Warn("Updated Condition to decommissioning started")
+
+		// TODO: add support for additional decommissioning target types.
+		// these are the steps that will take places between Started and Completed status
+
+		if err := m.updateWithCompletedCondition(ctx); err != nil {
+			return fmt.Errorf("failed to update decommission completed status: %w", err)
```

Comment:
The same is true here. The point of returning the error from the method is not to exit the decommission flow, right? If the device isn't connected to the service, do we still want to continue to decommission?

Reply:
Updated